src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -838,14 +837,17 @@ void Simulator::poll_for_MAVLink_messages(bool publish)
 				for (int i = 0; i < len; ++i) {
 					if (mavlink_parse_char(MAVLINK_COMM_1, serial_buf[i], &msg, &serial_status)) {
 						// have a message, handle it
-						handle_message(&msg, true);
+						set_publish(true);
+						handle_message(&msg);
```

Comment:
You should do it like this:
```
						bool prev_publish = _publish;
						_publish = true;
						handle_message(&msg);
						_publish = prev_publish;
```
Otherwise you're affecting and overriding the state for the next loop iteration.

Reply:
Thanks @bkueng ,
Changes requested are pushed up!

>Otherwise you're affecting and overriding the state for the next loop iteration.
You are correct, that was intentional.  The current logic in master cannot allow a persistent `publish == true` state to exist outside of one loop. This was a guard against that new possibility.

Let me know if you spot anything else I can address.  Thanks for your feedback!